Introduce Scope::getMaybeDefinedVariables

ruudk opened the pull request:
Fixes phpstan/phpstan#11772

Where would be the best way to create a test for this?

ruudk committed:
Introduce Scope::getMaybeDefinedVariables

Fixes phpstan/phpstan#11772

ruudk commented:
Every time 🙈

staabm commented:
I think a test could fit in

https://github.com/phpstan/phpstan-src/blob/1.12.x/tests/PHPStan/Analyser/ScopeTest.php

ruudk committed:
Add `@covers` to ScopeTest

By doing this, when you are inside MutatingScope you can press "Go to test" in PHPStorm and it
navigates you right to this test.

ruudk committed:
Add tests for getDefinedVariables and getMaybeDefinedVariables

ruudk commented:
@staabm Thanks. Somehow I could not find it when using `Go to test` in PHPStorm. I fixed that now.

The tests are added.

@ondrejmirtes This is ready for review!

ondrejmirtes commented:
Thank you!

janedbal commented on src/Analyser/MutatingScope.php:
```diff
@@ -574,6 +574,27 @@ public function getDefinedVariables(): array
 
+	/**
+	 * @api
+	 * @return array<int, string>
```
why not `list<string>` ?

ruudk replied:
Copied from getDefinedVars

janedbal replied:
Fixed: https://github.com/phpstan/phpstan-src/pull/3530/files

ruudk replied:
Great 🎉
